compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -93,12 +93,14 @@ object DesugarEnums {
    *       }
    */
   private def enumScaffolding(using Context): List[Tree] = {
+    import dotty.tools.dotc.transform.SymUtils._
+    val rawEnumClassRef = rawRef(enumClass.typeRef)
+    extension (tpe: NamedType) def ofRawEnum = AppliedTypeTree(ref(tpe), rawEnumClassRef)
     val valuesDef =
-      DefDef(nme.values, Nil, Nil, TypeTree(defn.ArrayOf(enumClass.typeRef)), Select(valuesDot(nme.values), nme.toArray))
+      DefDef(nme.values, Nil, Nil, defn.ArrayType.ofRawEnum, Select(valuesDot(nme.values), nme.toArray))
```

Comment:
We should have testcases checking that values work as expected both with higher-kinded and non-higher-kinded type parameters

Reply:
added tests